core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -158,21 +161,36 @@ private void tryStart() {
             containerIsStarting(containerInfo);
 
             // Wait until the container is running (may not be fully started)
-            profiler.start("Wait until container state=running");
-            Unreliables.retryUntilTrue(30, TimeUnit.SECONDS, () -> {
+            profiler.start("Wait until container state=running, or there's evidence it failed to start.");
+            final Boolean[] startedOK = {null};
```

Comment:
instead of a boolean you can define some non-Exception throwable (i.e. `ContainerStartError extends Throwable`) and catch it.